lib/widgets/message_list.dart

```diff
@@ -24,6 +24,158 @@ import 'edit_state_marker.dart';
 import 'text.dart';
 import 'theme.dart';
 
+/// Message-list styles that differ between light and dark themes.
+class MessageListTheme extends ThemeExtension<MessageListTheme> {
+  MessageListTheme.light() :
+    this._(
+      dateSeparator: Colors.black,
+      dateSeparatorText: const HSLColor.fromAHSL(0.75, 0, 0, 0.15).toColor(),
+      dmRecipientHeaderBg: const HSLColor.fromAHSL(1, 46, 0.35, 0.93).toColor(),
+      editedMovedMarkerCollapsed: const Color.fromARGB(128, 146, 167, 182),
+      messageTimestamp: const HSLColor.fromAHSL(0.8, 0, 0, 0.2).toColor(),
+      recipientHeaderText: const HSLColor.fromAHSL(1, 0, 0, 0.15).toColor(),
+      senderBotIcon: const HSLColor.fromAHSL(1, 180, 0.08, 0.65).toColor(),
+      senderName: const HSLColor.fromAHSL(1, 0, 0, 0.2).toColor(),
+      streamMessageBgDefault: Colors.white,
+      streamRecipientHeaderChevronRight: Colors.black.withOpacity(0.3),
+
+      // From the Figma mockup at:
+      //   https://www.figma.com/file/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=132-9684
+      // See discussion about design at:
+      //   https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/flutter.3A.20unread.20marker/near/1658008
+      // (Web uses a left-to-right gradient from hsl(217deg 64% 59%) to transparent,
+      // in both light and dark theme.)
+      unreadMarker: const HSLColor.fromAHSL(1, 227, 0.78, 0.59).toColor(),
+
+      unreadMarkerGap: Colors.white.withOpacity(0.6),
+
+      // TODO(design) this seems ad-hoc; is there a better color?
+      unsubscribedStreamRecipientHeaderBg: const Color(0xfff5f5f5),
+    );
+
+  MessageListTheme.dark() :
+    this._(
+      dateSeparator: Colors.white,
+      dateSeparatorText: const HSLColor.fromAHSL(0.75, 0, 0, 1).toColor(),
+      dmRecipientHeaderBg: const HSLColor.fromAHSL(1, 46, 0.15, 0.2).toColor(),
+      // TODO(#95) need proper dark-theme color (this is ad hoc)
+      editedMovedMarkerCollapsed: const Color.fromARGB(128, 214, 202, 194),
```

Comment:
For this and any similar spots that come up, I think the strategy will probably be:
* Pick something reasonable, as you did here.
* When dark theme is otherwise ready, ship it, and close #95.
* File a follow-up issue to track Vlad looking around in the app for anything he'd like to tweak about dark theme, and us applying those changes. In that issue, mention this and any other such items as areas for him to particularly look at.